### PR TITLE
feature/no macros

### PR DESCRIPTION
**Pull request by sonic182**
- **docs**
- **no metaprogramming**

### CODE REVIEW

**Review by mmacia — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -27,15 +27,21 @@ To create a basic chatbot using LlmComposer, you need to define a module that us
...
+
+  def simple_chat(msg) do
+    LlmComposer.simple_chat(@settings, msg)
+  end
 end
 
 {:ok, res} = MyChat.simple_chat("hi")
 
-IO.inspect res.main_response
+IO.inspect(res.main_response)
```
Comment (hectorperez) [inline README.md:44]: It would be handy to include an example of response.

File: README.md [hunk 2]
```diff
@@ -155,9 +166,9 @@ defmodule MyChat do
   end
 end
 
-{:ok, res} = MyChat.simple_chat("hi, how mutch is 1 + 2?")
+{:ok, res} = MyChat.simple_chat("hi, how much is 1 + 2?")
 
-IO.inspect res.main_response
+IO.inspect(res.main_response)
```
Comment (hectorperez) [inline README.md:171]: Same. Would `res.main_response` return "3"? It would be handy to add the response to the readme.

**Review by victor23k — APPROVED** [review]

File: lib/llm_composer.ex [hunk 3]
```diff
@@ -1,3 +1,129 @@
...
+          {:ok, res}
+        end
+
+      {:error, data} = resp ->
+        Logger.error("error in llm call: #{inspect(data)}")
+        resp
+    end)
+  end
+
+  @doc false
```
Comment (hectorperez) [inline lib/llm_composer.ex:107]: Do we need to add `@doc false` in private functions?

File: lib/llm_composer.ex [hunk 4]
```diff
@@ -1,3 +1,129 @@
...
+    user_prompt_prefix: "",
+    auto_exec_functions: false,
+    functions: []
+  }
+
+  # Initiate a simple chat interaction with the defined settings
+  {:ok, response} = LlmComposer.simple_chat(settings, "Hello, how are you?")
+
+  # Print the main response from the assistant
+  IO.inspect(response.main_response)
```
Comment (hectorperez) [inline lib/llm_composer.ex:26]: Again, a response could be handy.

**Review by hectorperez — CHANGES_REQUESTED** [review]
Brilliant! :100:  Thanks for removing the metaprogramming.

A minor suggestion below that would clarify the output. Thanks!

We could also add to the readme that there could be breaking changes. Something like this:
https://github.com/lebrunel/shifts?tab=readme-ov-file#current-dev-status

File: README.md (hunk 1 above)
Comment (sonic182) [reply README.md:44]: It is some lines below

File: README.md (hunk 2 above)
Comment (sonic182) [reply README.md:171]: Same, it is some lines below

File: lib/llm_composer.ex (hunk 3 above)
Comment (sonic182) [reply lib/llm_composer.ex:107]: I think not, gpt-4o did that for me, gonna remove it

File: README.md (hunk 2 above)
Comment (sonic182) [reply README.md:171]: Info is a bit below

**Pull request merged**